### ISSUE TITLE
I have some questions about functions : prob_to_QPSK_symbol

### ISSUE THREAD

**Issue by gaze-wu** [tomerraviv95/facilitating-adaptation-deep-receivers#1]
Dear Tomarraviv：
 I am learning your code. I have some questions about functions : prob_to_QPSK_symbol 

def prob_to_QPSK_symbol(p: torch.Tensor) -> torch.Tensor:
    """
    Converts Probabilities to QPSK Symbols by hard threshold.
    first bit: [0,0.5] -> '+1',[0.5,1] -> '-1'
    second bit: [0,0.5] -> '+1',[0.5,1] -> '-1'
    """
    p_real_neg = p[:, :, 0] + p[:, :, 2]
    first_symbol = (-1) * torch.sign(p_real_neg - HALF)
    p_img_neg = p[:, :, 1] + p[:, :, 2]
    second_symbol = (-1) * torch.sign(p_img_neg - HALF)
    s = torch.cat([first_symbol.unsqueeze(-1), second_symbol.unsqueeze(-1)], dim=-1)
    return torch.view_as_complex(s)

My question is why did you choose this decoding method, instead of directly using the QPSK symbol represented by the class index corresponding to the highest probability of the four classes output by DeepSIC?
Just like:
        0 -> 00
        2 -> 01
        1 -> 10
        3 -> 11
       return symbol = argmax(prob_qpsk)

Isn't this more convenient? Is there any reason for this design？

**Comment by tomerraviv95** [comment]
Thanks for the question gaze-wu,

It was only because I adopted uniform definitions at the transmitter and the receiver, where the transmitted symbols as +-1 +-1j and the transmitted labels before modulation as 1,2,3,4. As you can see in the compute_output function of the DeepSICTrainer class, I first compute the former and then demodulate them into the labels. In the data generation it is the opposite.

This reason is non-functional and mostly for readiblity, but I understand why it is confusing so thanks for the question again. You can obviously do as suggested in your code with no effect on error-rate performance.


Let me know if you have any more questions

**Comment by tomerraviv95** [comment]
Let me know if you have any more questions
